core/types/calls.cc

```diff
@@ -2129,33 +2138,20 @@ class Magic_selfNew : public IntrinsicMethod {
         }
         CallLocs sendLocs{args.locs.file, args.locs.call, args.locs.args[0], sendArgLocs};
 
-        TypePtr returnTy;
-        DispatchResult dispatched;
-        if (!self.data(gs)->isSingletonClass(gs)) {
-            // In the case that `self` is not a singleton class, we know that
-            // this was a call to `new` outside of a self context. Dispatch to
-            // an instance method named new, and see what happens.
-            DispatchArgs innerArgs{Names::new_(), sendLocs,   numPosArgs,
-                                   sendArgStore,  selfTy,     selfTy,
-                                   selfTy,        args.block, args.originForUninitialized};
-            dispatched = selfTy.dispatchCall(gs, innerArgs);
-            returnTy = dispatched.returnType;
-        } else {
-            // Otherwise, we know that this is the proper new intrinsic, and we
-            // should be returning something of type `T.attached_class`
+        DispatchArgs innerArgs{Names::new_(), sendLocs,   numPosArgs,
+                               sendArgStore,  selfTy,     selfTy,
+                               selfTy,        args.block, args.originForUninitialized};
+        auto dispatched = selfTy.dispatchCall(gs, innerArgs);
```

Comment:
This is a really nice simplification -- going via the existing new intrinsic makes a lot of sense.